### PR TITLE
Br 39 move logic of autowiring via method and fields from bean factory to autowiring bean post processor

### PR DESCRIPTION
**Pull request by ivanbolshak**
### Issue/Task

_[ List of issues or tasks in the form Task-xxx, Issue-xxx.]_

### Description

_[ Description of the changes for this PR.]_

### Testing

_[ Description of what testing was done. ]_

### Related Reviews

_[ Links to other PRs]_

### Screenshots

_[ Put your diagrams or screenshots with some highlights]_

### CODE REVIEW

File: src/main/java/svydovets/core/context/beanFactory/command/CommandFactoryEnum.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package svydovets.core.context.beanFactory.command;
+
+public enum CommandFactoryEnum {
```
Comment (Pan14ek) [inline src/main/java/svydovets/core/context/beanFactory/command/CommandFactoryEnum.java:3]: Let's give another name for this enum. `CommandName`, `CommandFunctionName`?

File: src/main/java/svydovets/core/context/beanFactory/command/CommandFactory.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+public class CommandFactory {
+
+    private Map<CommandFactoryEnum, Function<Class<?>, ?>> commands = new HashMap<>();
+
+    public void registryCommand(CommandFactoryEnum key, Function<Class<?>, ?> command){
+        commands.put(key, command);
+    }
+
+    public Function<Class<?>, ?> execute(CommandFactoryEnum key){
```
Comment (Pan14ek) [inline src/main/java/svydovets/core/context/beanFactory/command/CommandFactory.java:20]: I think we can replace wildcard with Object. What do you think?

File: src/main/java/svydovets/core/bpp/AutowiredAnnotationBeanPostProcessor.java [hunk 3]
```diff
@@ -1,14 +1,95 @@
...
+        populateProperties(bean);
         return BeanPostProcessor.super.postProcessBeforeInitialization(bean, beanName);
     }
 
     @Override
     public Object postProcessAfterInitialization(Object bean, String beanName) {
         return BeanPostProcessor.super.postProcessAfterInitialization(bean, beanName);
     }
+
+    public void populateProperties(Object bean) {
```
Comment (Pan14ek) [inline src/main/java/svydovets/core/bpp/AutowiredAnnotationBeanPostProcessor.java:41]: let's change public to private

File: src/main/java/svydovets/core/context/beanFactory/command/CommandFactory.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/*
+ * This file is a subject to the terms and conditions defined in
+ * file 'LICENSE.txt', which is part of this source code package.
+ */
```
Comment (Pan14ek) [inline src/main/java/svydovets/core/context/beanFactory/command/CommandFactory.java:4]: I am not sure that we have some licenses ;)

File: src/main/java/svydovets/core/context/beanFactory/command/CommandFactoryEnum.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+    FC_GET_BEAN, FC_GET_BEANS_OF_TYPE;
```
Comment (IgorZaporozhtsev) [inline src/main/java/svydovets/core/context/beanFactory/command/CommandFactoryEnum.java:4]:
```suggestion
    FC_GET_BEAN, FC_GET_BEANS_BY_TYPE;
```

File: src/main/java/svydovets/core/context/beanFactory/BeanFactory.java [hunk 6]
```diff
@@ -34,12 +32,20 @@
...
-    private final List<BeanPostProcessor> beanPostProcessors = new ArrayList<>(List.of(new AutowiredAnnotationBeanPostProcessor()));
+    private final List<BeanPostProcessor> beanPostProcessors = new ArrayList<>();
     private final PackageScanner packageScanner = new PackageScanner();
     private final BeanDefinitionFactory beanDefinitionFactory = new BeanDefinitionFactory();
 
+    private CommandFactory commandFactory = new CommandFactory();
+
+    public BeanFactory() {
+        commandFactory.registryCommand(CommandFactoryEnum.FC_GET_BEAN, this::getBean);
+        commandFactory.registryCommand(CommandFactoryEnum.FC_GET_BEANS_OF_TYPE, this::getBeansOfType);
```
Comment (IgorZaporozhtsev) [inline src/main/java/svydovets/core/context/beanFactory/BeanFactory.java:43]:
```suggestion
        commandFactory.registryCommand(CommandFactoryEnum.FC_GET_BEANS_OF_TYPE, this::getBeansByType);
```

Comment (ivanbolshak) [reply src/main/java/svydovets/core/context/beanFactory/BeanFactory.java:43]: @IgorZaporozhtsev  this is method reference to BeanFactory, firstly have to be renamed there

File: src/main/java/svydovets/core/context/beanFactory/command/CommandFactoryEnum.java (hunk 5 above)
Comment (ivanbolshak) [reply src/main/java/svydovets/core/context/beanFactory/command/CommandFactoryEnum.java:4]: @IgorZaporozhtsev named the same like appropriate method in BeanFactory

**Review by Pan14ek — APPROVED** [review]

**Pull request merged**